tests/test_abspath.py

```diff
@@ -230,14 +230,7 @@ def test_abspath_write(local_test_path):
 
 def test_abspath_write_no_permission():
     try:
-        u = AbsPath('/x.tmp')
-        u.write('test')
-        assert False
-    except PermissionError:
-        pass
-
-    try:
-        u = AbsPath('/x.tmp')
+        u = AbsPath('/test-permission-denied/x.tmp')
```

Comment:
What you are saying here is you expect a `PermissionError` right? There is a built in mechanism for this in pytest, that will make the intention of your test clear and you will not need to use this workaround. For example this test can be written like:
 ```
def test_abspath_write_no_permission():
    u = AbsPath("/test-permission-denied/x.tmp")
    with pytest.raises(PermissionError):
        u.write("test")
```
For more information, see: https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions

Reply:
Thanks for this. This looks very useful. I have been using `@pytest.mark.xfail(raises=)` and `try-except` to catch an expected error. I will replace all of them with this new context-based method.